impl/src/testing.rs

```diff
@@ -56,3 +96,107 @@ pub fn make_workspace(
 pub fn make_basic_workspace() -> (TempDir, CargoWorkspaceFiles) {
   make_workspace(basic_toml(), Some(basic_lock()))
 }
+
+/** Creates mock responses for fetching crate metadata and source code from a crate registry */
```

Comment:
Nit: More precise would be "Configures the given mock_server (representing a crates.io remote) to return mock responses for the given crate and version"